File: app/src/main/java/com/bth/running/running/RunningManager.java
package com.bth.running.running;

import android.location.Location;

/**
 * @author Martin Macheiner
 *         Date: 05.09.2017.
 */

public interface RunningManager {

    void startRunRecording(long startTimeInMillis);

    void stopRunRecord();

    Run updateCurrentRun(Location location);

    Run getCurrentRun();

    Run getFinishedRun(boolean resetRun);

    boolean isRecording();

}

File: app/src/main/java/at/shockbytes/corey/running/running/RunningManager.kt
package at.shockbytes.corey.running.running

import android.location.Location

/**
 * @author  Martin Macheiner
 * Date:    05.09.2017.
 */

interface RunningManager {

    val currentRun: Run

    val isRecording: Boolean

    fun startRunRecording(startTimeInMillis: Long)

    fun stopRunRecord()

    fun updateCurrentRun(location: Location): Run

    fun getFinishedRun(resetRun: Boolean): Run

}
